src/opera/pge/dist_s1/dist_s1_pge.py

```diff
@@ -31,12 +33,226 @@ class DistS1PreProcessorMixin(PreProcessorMixin):
+
+        if len(pre_rtc_copol) != len(pre_rtc_crosspol) or len(post_rtc_copol) != len(post_rtc_crosspol):
+            msg = (f'Lengths of input pre/post co/cross pol input RTC lists differ: pre {len(pre_rtc_copol)} '
+                   f'{len(pre_rtc_crosspol)} post {len(post_rtc_copol)} {len(post_rtc_crosspol)}')
+            self.logger.critical(
+                self.name,
+                ErrorCode.INVALID_INPUT,
+                msg
+            )
+
+    def __validate_rtc_lists_are_pge_subset(self):
+        """Ensures all RTCs in the SAS group are also in the PGE group"""
+        sas_config = self.runconfig.sas_config
+
+        pre_rtc_copol = sas_config["run_config"]["pre_rtc_copol"]
+        pre_rtc_crosspol = sas_config["run_config"]["pre_rtc_crosspol"]
+        post_rtc_copol = sas_config["run_config"]["post_rtc_copol"]
+        post_rtc_crosspol = sas_config["run_config"]["post_rtc_crosspol"]
+
+        all_rtcs = pre_rtc_copol + pre_rtc_crosspol + post_rtc_copol + post_rtc_crosspol
+
+        if not set([basename(rtc) for rtc in all_rtcs]).issubset([basename(rtc) for rtc in self.runconfig.input_files]):
+            msg = 'SAS RTC file groups do not make a subset of PGE Input Files'
+            self.logger.critical(
+                self.name,
+                ErrorCode.INVALID_INPUT,
+                msg
+            )
+
+    def __validate_rtc_filenames(self):
+        """
+        Validates that all input RTCs have the standard filename defined in the RTC product specification.
+
+        See: https://www.jpl.nasa.gov/go/opera/products/rtc-product/
+
+        Returns
+        -------
+        <rtc_matches>: List of re.Match objects
+            List of RTC filenames matched by a regular expression. Used in later validation. Only returns
+            if this validation is passing.
+        """
+        sas_config = self.runconfig.sas_config
+
+        pre_rtc_copol = sas_config["run_config"]["pre_rtc_copol"]
+        pre_rtc_crosspol = sas_config["run_config"]["pre_rtc_crosspol"]
+        post_rtc_copol = sas_config["run_config"]["post_rtc_copol"]
+        post_rtc_crosspol = sas_config["run_config"]["post_rtc_crosspol"]
+
+        all_rtcs = pre_rtc_copol + pre_rtc_crosspol + post_rtc_copol + post_rtc_crosspol
+
+        rtc_matches = [self._rtc_pattern.match(basename(rtc)) for rtc in all_rtcs]
+        mismatches = [rtc for match, rtc in zip(rtc_matches, all_rtcs) if match is None]
+
+        if len(mismatches) > 0:
+            msg = f'Invalid RTC filenames in SAS input: {mismatches}'
+            self.logger.critical(
+                self.name,
+                ErrorCode.INVALID_INPUT,
+                msg
+            )
+
+        return rtc_matches
+
+    def __validate_rtc_homogeneity(self, rtc_matches):
+        """
+        Validates all RTCs are from either Sentinel-1A or Sentinel-1C.
+
+        Parameters
+        ----------
+        <rtc_matches>: List of re.Match objects
```

Comment:
Why the brackets here?

Reply:
I copied the formatting for the params section of a docstring from elsewhere... I'll remove them